linter.py

```diff
@@ -10,6 +26,48 @@ class XO(NodeLinter):
 		r' \((?P<code>.+)\)$'
 	)
 	defaults = {
-		'selector': 'source.js - meta.attribute-with-value',
+		'selector': OPTIMISTIC_SELECTOR,
 		'disable_if_not_dependency': True
 	}
+
+	def run(self, cmd, code):
+		self.ensure_plugin_installed()
+		return super().run(cmd, code)
+
+	def ensure_plugin_installed(self) -> bool:
+		# If the user changed the selector, we take it as is
+		if self.settings['selector'] != OPTIMISTIC_SELECTOR:
+			return True
+
+		# Happy path.
+		if self.view.match_selector(0, STANDARD_SELECTOR):
+			return True
+
+		# If we're here we must be pessimistic.
+
+		# The 'project_root' has the relevant 'package.json' file colocated.
+		# If we fallback to a global installation there is no 'project_root',
+		# t.i. no auto-selector in that case as well.
+		project_root = self.context.get('project_root')
+		if project_root:
+			# We still need to be careful, in case SL deduced a 'project_root'
+			# without checking for the 'package.json' explicitly. Basically, a
+			# happy path for SL core.
+			manifest_file = os.path.join(project_root, 'package.json')
+			try:
+				manifest = read_json_file(manifest_file)
+			except Exception:
+				pass
+			else:
+				defined_plugins = PLUGINS.keys() & (
+					manifest.get('dependencies', {}).keys()
+					| manifest.get('devDependencies', {}).keys()
+				)
+				selector = ', '.join(PLUGINS[name] for name in defined_plugins)
+				if selector and self.view.match_selector(0, selector):
+					return True
+
+		# Indicate an error which usually can only be solved by changing
+		# the environment. Silently, do not notify and disturb the user!
```

Comment:
What do you mean by changing the environment?

Reply:
By installing a plugin or changing the `selector` manually. T.i. something we don't control or can fix.